app/models/section.rb

```diff
@@ -0,0 +1,94 @@
+class Section
+  include ActiveModel::Model
+  include ActiveModel::Validations
+
+  attr_accessor :content_id, :base_path, :title, :summary, :body, :updated_at, :public_updated_at, :organisations, :update_type
+
+  validates :summary, presence: true
+  validates :title, presence: true
+  validates :body, presence: true, safe_html: true
+
+  def initialize(params)
+    @content_id = params.fetch(:content_id, SecureRandom.uuid)
+    @title = params.fetch(:title, nil)
+    @summary = params.fetch(:summary, nil)
+    @body = params.fetch(:body, nil)
+    @publication_state = params.fetch(:publication_state, nil)
+    @updated_at = params.fetch(:updated_at, nil)
+    @public_updated_at = params.fetch(:public_updated_at, nil)
+    @manual = params.fetch(:manual_content_id)
+  end
+
+  %w{draft live redrafted}.each do |state|
+    define_method("#{state}?") do
+      publication_state == state
+    end
+  end
+
+  def updated_at
+    @updated_at ||= Time.zone.now
+  end
+
+  def updated_at=(timestamp)
+    @updated_at = Time.parse(timestamp.to_s) unless timestamp.nil?
+  end
+
+  def public_updated_at
+    @public_updated_at ||= Time.zone.now
+  end
+
+  def public_updated_at=(timestamp)
+    @public_updated_at = Time.parse(timestamp.to_s) unless timestamp.nil?
+  end
+
+  def manual=(manual_content_id)
```

Comment:
You're right that does make more sense. What do the setter methods for the model become then?